keps/sig-node/3008-qos-class-resources/README.md

```diff
@@ -0,0 +1,2314 @@
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+In this section we refer to different
+[implementation phases](#implementation-phases). In this KEP we're now
+targeting phase 1.
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: QOSResources
+  - Components depending on the feature gate:
+    - Implementation Phase 1:
+        - kubelet
+        - kube-apiserver
+        - kube-scheduler
+    - Future phases (with quota and limitranges support):
+        - kubelet
+        - kube-apiserver
+        - kube-scheduler
+        - kube-controller-manager
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+Yes it can.
+
+Running workloads continue to work without any changes. Restarting or
+re-deploying a workload causes it to fail as the requested QoS-class resources
+are not available in Kubernetes anymore. The resources are still supported by
+the underlying runtime but disabling the feature in Kubernetes makes them
```

Comment:
By "Kubernetes" do you mean kube-apiserver, kubelet or both?

Reply:
I think the answers are:
- kubelet - it would ignore this
- kube-scheduler - ignore it
- api-server - reject except in the case of an update where they fields were already present and are unchanged, in which case it is preserved (see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md#lifecycle).


Even with these unresoloved, I consider this non-blocking for PRR, but of course you must resolve this before code freeze.

Reply:
I updated this section with a more detailed description on the behavior of individual components (kubelet, apiserver, scheduler).

Reply:
ping @logicalhan ^